src/mongocxx/include/mongocxx/v_noabi/mongocxx/search_index_model.hpp

```diff
@@ -25,6 +25,11 @@ class search_index_model {
     search_index_model(bsoncxx::v_noabi::document::view_or_value definition);
     search_index_model(bsoncxx::v_noabi::string::view_or_value name,
                        bsoncxx::v_noabi::document::view_or_value definition);
+    search_index_model(bsoncxx::v_noabi::document::view_or_value definition,
+                       bsoncxx::v_noabi::string::view_or_value type);
+    search_index_model(bsoncxx::v_noabi::string::view_or_value name,
+                       bsoncxx::v_noabi::document::view_or_value definition,
+                       bsoncxx::v_noabi::string::view_or_value type);
```

Comment:
Rather than add overloads, suggest adding a setter:
```c++
///
/// Sets type of a search_index_model.
///
/// @param type The type for this index. Can be either "search" or "vectorSearch".
///
/// @return
///   A reference to this object to facilitate method chaining.
search_index_model& type(bsoncxx::v_noabi::string::view_or_value type);
```

If future options are specified in `SearchIndexModel`, following the pattern of adding overloads may double the needed overloads.

The setter follows other patterns in the driver for setting options ([example](https://github.com/mongodb/mongo-cxx-driver/blob/5b30e19186b2bddc731b5b20436adf5641e0f169/src/mongocxx/include/mongocxx/v_noabi/mongocxx/model/update_one.hpp#L105-L114)).